packages/data/addon/services/navi-metadata.ts

```diff
@@ -0,0 +1,168 @@
+
+      const normalized = this.serializerFor(dataSourceType).normalize('everything', payload, dataSourceName);
+      if (normalized) {
+        this.loadEverythingMetadataIntoKeg(normalized, dataSourceName);
+      }
+    }
+    //TODO store promises, so two requests for the same metadata use the same promise
+  }
+
+  /**
+   * Provides an array of all loaded models for a given type
+   * @param type - model type
+   * @param dataSourceName - optional name of data source
+   */
+  all<K extends keyof MetadataModelRegistry>(type: K, dataSourceName?: string): EmberArray<MetadataModelRegistry[K]> {
+    assert(
+      `Metadata must have the requested data source loaded: ${dataSourceName}`,
+      !dataSourceName || this.loadedDataSources.has(dataSourceName)
+    );
+    return this.keg.all(`metadata/${type}`, dataSourceName) as EmberArray<MetadataModelRegistry[K]>;
+  }
+
+  /**
+   * Provides a single loaded model instance given an identifier
+   * @param type - model type
+   * @param id - identifier value of model
+   * @param dataSourceName - name of data source
+   */
+  getById<K extends keyof MetadataModelRegistry>(
+    type: K,
+    id: string,
+    dataSourceName: string
+  ): MetadataModelRegistry[K] | undefined {
+    assert('`dataSourceName` argument required', dataSourceName);
+    return this.keg.getById(`metadata/${type}`, id, dataSourceName) as MetadataModelRegistry[K];
+  }
+
+  /**
+   * Fetches from data source a single model instance given an identifier
+   * @param type - model type
+   * @param id - identifier value of model
+   * @param dataSourceName - name of data source
+   */
+  async fetchById<K extends keyof MetadataModelRegistry>(
+    type: K,
+    id: string,
+    dataSourceName: string
+  ): Promise<MetadataModelRegistry[K] | undefined> {
+    assert('`dataSourceName` argument required', dataSourceName);
+    const { type: dataSourceType } = this.dataSourceFor(dataSourceName);
+    const rawPayload = await this.adapterFor(dataSourceType).fetchById(type, id, { dataSourceName });
+    const normalized = rawPayload
+      ? this.serializerFor(dataSourceType).normalize(type, rawPayload, dataSourceName)
+      : undefined;
+
+    if (normalized) {
+      const model = this.loadMetadataForType(type, normalized, dataSourceName);
+      return model?.firstObject as MetadataModelRegistry[K];
+    }
+    return undefined;
+  }
+
+  /**
+   * Provides a single loaded model instance or fetches one, given an identifier
+   * @param type - model type
+   * @param id - identifier value of model
+   * @param dataSourceName - name of data source
+   */
+  findById<K extends keyof MetadataModelRegistry>(
+    type: K,
+    id: string,
+    dataSourceName: string
+  ): Promise<MetadataModelRegistry[K] | undefined> {
+    assert('`dataSourceName` argument required', dataSourceName);
+    const kegRecord = this.keg.getById(`metadata/${type}`, id, dataSourceName);
+    if (kegRecord && !kegRecord.partialData) {
```

Comment:
`partialData` was removed from `KegRecord` right?

Reply:
Right, but it is a thing for now